### ISSUE TITLE
Netty3TransportMultiPortIntegrationIT.testThatTransportClientCanConnect fails to connect sometimes in CI

### ISSUE THREAD

**Issue by nik9000** [elastic/elasticsearch#24395]
[This](https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+5.3+multijob-unix-compatibility/os=amazon/153/console) build found a failure to connect with a timeout. Here is the failure:
```
  1> [2017-04-28T10:13:21,286][INFO ][o.e.t.n.Netty3TransportMultiPortIntegrationIT] [testThatTransportClientCanConnect]: before test
  1> [2017-04-28T10:13:21,288][INFO ][o.e.t.n.Netty3TransportMultiPortIntegrationIT] [Netty3TransportMultiPortIntegrationIT#testThatTransportClientCanConnect]: setting up test
  1> [2017-04-28T10:13:21,290][INFO ][o.e.t.InternalTestCluster] Setup InternalTestCluster [SUITE-CHILD_VM=[0]-CLUSTER_SEED=[-4541300959115935969]-HASH=[D7EBD24AC50ED]-cluster] with seed [C0FA0ED6A9C0FF1F] using [0] dedicated masters, [1] (data) nodes and [0] coord only nodes (min_master_nodes are [auto-managed])
  1> [2017-04-28T10:13:21,293][INFO ][o.e.n.Node               ] [node_s0] initializing ...
  1> [2017-04-28T10:13:21,299][INFO ][o.e.e.NodeEnvironment    ] [node_s0] using [2] data paths, mounts [[/ (/dev/xvda1)]], net usable_space [386.1gb], net total_space [492gb], spins? [unknown], types [ext4]
  1> [2017-04-28T10:13:21,299][INFO ][o.e.e.NodeEnvironment    ] [node_s0] heap size [491mb], compressed ordinary object pointers [true]
  1> [2017-04-28T10:13:21,300][INFO ][o.e.n.Node               ] [node_s0] node name [node_s0], node ID [OuVxcU5KTv6IdF1_2E3ZcQ]
  1> [2017-04-28T10:13:21,300][INFO ][o.e.n.Node               ] [node_s0] version[5.3.2-SNAPSHOT], pid[19483], build[f5205a1/2017-04-28T13:43:35.102Z], OS[Linux/4.4.44-39.55.amzn1.x86_64/amd64], JVM[Oracle Corporation/OpenJDK 64-Bit Server VM/1.8.0_121/25.121-b13]
  1> [2017-04-28T10:13:21,300][WARN ][o.e.n.Node               ] [node_s0] version [5.3.2-SNAPSHOT] is a pre-release version of Elasticsearch and is not suitable for production
  1> [2017-04-28T10:13:21,301][INFO ][o.e.p.PluginsService     ] [node_s0] no modules loaded
  1> [2017-04-28T10:13:21,301][INFO ][o.e.p.PluginsService     ] [node_s0] loaded plugin [org.elasticsearch.index.MockEngineFactoryPlugin]
  1> [2017-04-28T10:13:21,301][INFO ][o.e.p.PluginsService     ] [node_s0] loaded plugin [org.elasticsearch.search.MockSearchService$TestPlugin]
  1> [2017-04-28T10:13:21,301][INFO ][o.e.p.PluginsService     ] [node_s0] loaded plugin [org.elasticsearch.test.ESIntegTestCase$TestSeedPlugin]
  1> [2017-04-28T10:13:21,301][INFO ][o.e.p.PluginsService     ] [node_s0] loaded plugin [org.elasticsearch.test.discovery.TestZenDiscovery$TestPlugin]
  1> [2017-04-28T10:13:21,301][INFO ][o.e.p.PluginsService     ] [node_s0] loaded plugin [org.elasticsearch.test.store.MockFSIndexStore$TestPlugin]
  1> [2017-04-28T10:13:21,301][INFO ][o.e.p.PluginsService     ] [node_s0] loaded plugin [org.elasticsearch.transport.Netty3Plugin]
  1> [2017-04-28T10:13:21,322][WARN ][d.o.a.l.s.StoreRateLimiting] Store rate limiting is deprecated and will be removed in 6.0
  1> [2017-04-28T10:13:21,324][WARN ][o.e.d.t.n.Netty3Transport] transport type [netty3] is deprecated
  1> [2017-04-28T10:13:21,373][INFO ][o.e.n.Node               ] [node_s0] initialized
  1> [2017-04-28T10:13:21,374][INFO ][o.e.n.Node               ] [node_s0] starting ...
  1> [2017-04-28T10:13:21,383][INFO ][o.e.t.TransportService   ] [node_s0] publish_address {127.0.0.1:9440}, bound_addresses {127.0.0.1:9440}
  1> [2017-04-28T10:13:21,383][INFO ][o.e.t.TransportService   ] [node_s0] profile [client1]: publish_address {127.0.0.7:4321}, bound_addresses {127.0.0.1:49769}
  1> [2017-04-28T10:13:21,388][INFO ][o.e.t.d.MockZenPing      ] [node_s0] pinging using mock zen ping
  1> [2017-04-28T10:13:21,389][INFO ][o.e.c.s.ClusterService   ] [node_s0] new_master {node_s0}{OuVxcU5KTv6IdF1_2E3ZcQ}{LK5DG-fOTPCyThCkohmlAQ}{127.0.0.1}{127.0.0.1:9440}, reason: zen-disco-elected-as-master ([0] nodes joined)
  1> [2017-04-28T10:13:21,392][INFO ][o.e.n.Node               ] [node_s0] started
  1> [2017-04-28T10:13:21,398][INFO ][o.e.p.PluginsService     ] [transport_client_node_s0] no modules loaded
  1> [2017-04-28T10:13:21,398][INFO ][o.e.p.PluginsService     ] [transport_client_node_s0] loaded plugin [org.elasticsearch.transport.MockTcpTransportPlugin]
  1> [2017-04-28T10:13:21,398][INFO ][o.e.p.PluginsService     ] [transport_client_node_s0] loaded plugin [org.elasticsearch.transport.Netty3Plugin]
  1> [2017-04-28T10:13:21,406][WARN ][o.e.d.t.n.Netty3Transport] transport type [netty3] is deprecated
  1> [2017-04-28T10:13:21,412][INFO ][o.e.g.GatewayService     ] [node_s0] recovered [0] indices into cluster_state
  1> [2017-04-28T10:13:21,462][INFO ][o.e.t.n.Netty3TransportMultiPortIntegrationIT] test using _default_ mappings: [{"_default_":{}}]
  1> [2017-04-28T10:13:21,491][INFO ][o.e.t.n.Netty3TransportMultiPortIntegrationIT] [Netty3TransportMultiPortIntegrationIT#testThatTransportClientCanConnect]: all set up test
  1> [2017-04-28T10:13:21,493][INFO ][o.e.p.PluginsService     ] [_client_] no modules loaded
  1> [2017-04-28T10:13:21,493][INFO ][o.e.p.PluginsService     ] [_client_] loaded plugin [org.elasticsearch.transport.Netty3Plugin]
  1> [2017-04-28T10:13:21,500][WARN ][o.e.d.t.n.Netty3Transport] transport type [netty3] is deprecated
  1> [2017-04-28T10:13:21,516][INFO ][o.e.c.t.TransportClientNodesService] [_client_] failed to connect to node [{#transport#-1}{GLZlB-BERiyWiObN6RjJSQ}{127.0.0.1}{127.0.0.1:49768}], removed from nodes list
  1> org.elasticsearch.transport.ConnectTransportException: [][127.0.0.1:49768] connect_timeout[30s]
  1> 	at org.elasticsearch.transport.netty3.Netty3Transport.connectToChannels(Netty3Transport.java:378) ~[main/:?]
  1> 	at org.elasticsearch.transport.TcpTransport.openConnection(TcpTransport.java:495) ~[elasticsearch-5.3.2-SNAPSHOT.jar:?]
  1> 	at org.elasticsearch.transport.TcpTransport.connectToNode(TcpTransport.java:460) ~[elasticsearch-5.3.2-SNAPSHOT.jar:?]
  1> 	at org.elasticsearch.transport.TransportService.connectToNode(TransportService.java:314) ~[elasticsearch-5.3.2-SNAPSHOT.jar:?]
  1> 	at org.elasticsearch.client.transport.TransportClientNodesService$SimpleNodeSampler.doSample(TransportClientNodesService.java:408) ~[elasticsearch-5.3.2-SNAPSHOT.jar:5.3.2-SNAPSHOT]
  1> 	at org.elasticsearch.client.transport.TransportClientNodesService$NodeSampler.sample(TransportClientNodesService.java:354) ~[elasticsearch-5.3.2-SNAPSHOT.jar:5.3.2-SNAPSHOT]
  1> 	at org.elasticsearch.client.transport.TransportClientNodesService.addTransportAddresses(TransportClientNodesService.java:195) ~[elasticsearch-5.3.2-SNAPSHOT.jar:5.3.2-SNAPSHOT]
  1> 	at org.elasticsearch.client.transport.TransportClient.addTransportAddress(TransportClient.java:322) ~[elasticsearch-5.3.2-SNAPSHOT.jar:5.3.2-SNAPSHOT]
  1> 	at org.elasticsearch.transport.netty3.Netty3TransportMultiPortIntegrationIT.testThatTransportClientCanConnect(Netty3TransportMultiPortIntegrationIT.java:80) ~[test/:?]
  1> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
  1> 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
  1> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
  1> 	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_121]
  1> 	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1713) ~[randomizedtesting-runner-2.4.0.jar:?]
  1> 	at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:907) ~[randomizedtesting-runner-2.4.0.jar:?]
  1> 	at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:943) ~[randomizedtesting-runner-2.4.0.jar:?]
  1> 	at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:957) ~[randomizedtesting-runner-2.4.0.jar:?]
  1> 	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) ~[randomizedtesting-runner-2.4.0.jar:?]
  1> 	at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49) ~[lucene-test-framework-6.4.2.jar:6.4.2 34a975ca3d4bd7fa121340e5bcbf165929e0542f - ishan - 2017-03-01 23:23:14]
  1> 	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45) ~[lucene-test-framework-6.4.2.jar:6.4.2 34a975ca3d4bd7fa121340e5bcbf165929e0542f - ishan - 2017-03-01 23:23:14]
  1> 	at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48) ~[lucene-test-framework-6.4.2.jar:6.4.2 34a975ca3d4bd7fa121340e5bcbf165929e0542f - ishan - 2017-03-01 23:23:14]
  1> 	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64) ~[lucene-test-framework-6.4.2.jar:6.4.2 34a975ca3d4bd7fa121340e5bcbf165929e0542f - ishan - 2017-03-01 23:23:14]
  1> 	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47) ~[lucene-test-framework-6.4.2.jar:6.4.2 34a975ca3d4bd7fa121340e5bcbf165929e0542f - ishan - 2017-03-01 23:23:14]
  1> 	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) ~[randomizedtesting-runner-2.4.0.jar:?]
  1> 	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:367) ~[randomizedtesting-runner-2.4.0.jar:?]
  1> 	at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:811) ~[randomizedtesting-runner-2.4.0.jar:?]
  1> 	at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:462) ~[randomizedtesting-runner-2.4.0.jar:?]
  1> 	at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:916) ~[randomizedtesting-runner-2.4.0.jar:?]
  1> 	at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:802) ~[randomizedtesting-runner-2.4.0.jar:?]
  1> 	at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:852) ~[randomizedtesting-runner-2.4.0.jar:?]
  1> 	at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:863) ~[randomizedtesting-runner-2.4.0.jar:?]
  1> 	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45) ~[lucene-test-framework-6.4.2.jar:6.4.2 34a975ca3d4bd7fa121340e5bcbf165929e0542f - ishan - 2017-03-01 23:23:14]
  1> 	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) ~[randomizedtesting-runner-2.4.0.jar:?]
  1> 	at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41) ~[lucene-test-framework-6.4.2.jar:6.4.2 34a975ca3d4bd7fa121340e5bcbf165929e0542f - ishan - 2017-03-01 23:23:14]
  1> 	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40) ~[randomizedtesting-runner-2.4.0.jar:?]
  1> 	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40) ~[randomizedtesting-runner-2.4.0.jar:?]
  1> 	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) ~[randomizedtesting-runner-2.4.0.jar:?]
  1> 	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) ~[randomizedtesting-runner-2.4.0.jar:?]
  1> 	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) ~[randomizedtesting-runner-2.4.0.jar:?]
  1> 	at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53) ~[lucene-test-framework-6.4.2.jar:6.4.2 34a975ca3d4bd7fa121340e5bcbf165929e0542f - ishan - 2017-03-01 23:23:14]
  1> 	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47) ~[lucene-test-framework-6.4.2.jar:6.4.2 34a975ca3d4bd7fa121340e5bcbf165929e0542f - ishan - 2017-03-01 23:23:14]
  1> 	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64) ~[lucene-test-framework-6.4.2.jar:6.4.2 34a975ca3d4bd7fa121340e5bcbf165929e0542f - ishan - 2017-03-01 23:23:14]
  1> 	at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54) ~[lucene-test-framework-6.4.2.jar:6.4.2 34a975ca3d4bd7fa121340e5bcbf165929e0542f - ishan - 2017-03-01 23:23:14]
  1> 	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) ~[randomizedtesting-runner-2.4.0.jar:?]
  1> 	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:367) ~[randomizedtesting-runner-2.4.0.jar:?]
  1> 	at java.lang.Thread.run(Thread.java:745) ~[?:1.8.0_121]
  1> Caused by: java.net.ConnectException: Connection refused: /127.0.0.1:49768
  1> 	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method) ~[?:?]
  1> 	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717) ~[?:?]
  1> 	at org.jboss.netty.channel.socket.nio.NioClientBoss.connect(NioClientBoss.java:152) ~[netty-3.10.6.Final.jar:?]
  1> 	at org.jboss.netty.channel.socket.nio.NioClientBoss.processSelectedKeys(NioClientBoss.java:105) ~[netty-3.10.6.Final.jar:?]
  1> 	at org.jboss.netty.channel.socket.nio.NioClientBoss.process(NioClientBoss.java:79) ~[netty-3.10.6.Final.jar:?]
  1> 	at org.jboss.netty.channel.socket.nio.AbstractNioSelector.run(AbstractNioSelector.java:337) ~[netty-3.10.6.Final.jar:?]
  1> 	at org.jboss.netty.channel.socket.nio.NioClientBoss.run(NioClientBoss.java:42) ~[netty-3.10.6.Final.jar:?]
  1> 	at org.jboss.netty.util.ThreadRenamingRunnable.run(ThreadRenamingRunnable.java:108) ~[netty-3.10.6.Final.jar:?]
  1> 	at org.jboss.netty.util.internal.DeadLockProofWorker$1.run(DeadLockProofWorker.java:42) ~[netty-3.10.6.Final.jar:?]
  1> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) ~[?:1.8.0_121]
  1> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) ~[?:1.8.0_121]
  1> 	... 1 more
  1> [2017-04-28T10:13:21,529][INFO ][o.e.t.n.Netty3TransportMultiPortIntegrationIT] [Netty3TransportMultiPortIntegrationIT#testThatTransportClientCanConnect]: cleaning up after test
  1> [2017-04-28T10:13:21,548][INFO ][o.e.t.n.Netty3TransportMultiPortIntegrationIT] [Netty3TransportMultiPortIntegrationIT#testThatTransportClientCanConnect]: cleaned up after test
  1> [2017-04-28T10:13:21,548][INFO ][o.e.t.n.Netty3TransportMultiPortIntegrationIT] [testThatTransportClientCanConnect]: after test
```

I don't know this code super well and it doesn't reproduce.

**Comment by jaymode** [comment]
Another instance on the 5.5 branch https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+5.5+multijob-windows-compatibility/47

**Comment by danielmitterdorfer** [comment]
Another one also on 5.5: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+5.5+multijob-windows-compatibility/19

**Comment by jasontedor** [comment]
Closed by a7f164696ba939f9af58de6cc2da80ab9de432bf
